### PR TITLE
Avoid building a URI again and again in JarResource

### PR DESCRIPTION
**Pull request by gsmet**
@franz1981 could you have a look at this change? It's related to https://github.com/quarkusio/quarkus/pull/39447 .

### CODE REVIEW

**Comment by franz1981** [comment]
now checking @gsmet!

**Comment by franz1981** [comment]
adding @geoand too which is the one who wrote this one

I'm mostly checking there's some evident performance reason why it shouldn't be the same (which doesn't seem the case , at a first look)

**Comment by gsmet** [comment]
Shouldn't be the same as what?

**Comment by franz1981** [comment]
I mean, it should be the same as it was before (from a semantic perspective), but i'm double-checking if there is any performance unexpected reason why it is not

**Comment by franz1981** [comment]
So, in theory `JarResource::init` happens just once at https://github.com/quarkusio/quarkus/blob/c25f1db3d6d26e14762faadbfc6f7dd9b2af93c6/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/SerializedApplication.java#L165, meaning that there's not reason to cache `URI jarUri` and can just chain `jarPath.toUti().toURL()` on it, to save retaining more heap and necessary and let `URI` to be Gc'd.
I'm now checking if the allocated bytes are the same for the 2 methods.

**Comment by gsmet** [comment]
Have you looked at the call I removed? Because that was the whole purpose of this patch. Removing a call that we do a lot of times for the config stuff.

**Comment by franz1981** [comment]
uh @gsmet !

you mean https://github.com/quarkusio/quarkus/pull/39508/files#diff-49c6546297493b873346e7eb56a6f56a84964d13075679c56497b739674a97a0L113 ?
good point, I've missed it!

**Comment by franz1981** [comment]
TLDR for the getting-started quickstart I cannot see any measurable difference, because it seems `getResourceURL` is called just 9 times (i've attached the profiling data both for allocation(s) and number of times the method is called and where) 
[profling_data.zip](https://github.com/quarkusio/quarkus/files/14635556/profling_data.zip)

**Comment by gsmet** [comment]
OK, well, if it doesn't bring anything to the plate, we can close this.
